src/buffer/out/textBuffer.cpp

```diff
@@ -837,11 +838,10 @@ void TextBuffer::Reset()
         const SHORT TopRowIndex = (GetFirstRowIndex() + TopRow) % currentSize.Y;
 
         // rotate rows until the top row is at index 0
-        const ROW& newTopRow = _storage.at(TopRowIndex);
-        while (&newTopRow != &_storage.front())
+        for (int i = 0; i < TopRowIndex; i++)
         {
-            _storage.push_back(std::move(_storage.front()));
-            _storage.pop_front();
+            _storage.emplace_back(std::move(_storage.front()));
+            _storage.erase(_storage.begin());
         }

```

Comment:
This looks like a bulk operation in disguise. Given that there's no range-emplace (and a moving read iterator we could use to bulk-`move` the first N elements), would be it cheaper if we moved the first N elements in the loop and did a bulk `erase` outside of it? I reckon it would make the erase operation `O(N)` instead of `O(TopRowIndex*N)`

Reply:
I'm assuming vector's internal loading factor makes it acceptable to `emplace_back` one by one without using `reserve` upfront to add space for `TopRowIndex`.

Perhaps doing it this way/one-by-one keeps the vector from blowing up its reserved storage?

Reply:
Final thought: since we have to touch each member off the vector during erase, would it be overall cheaper to allocate a new vector, move elements TopRowIndex..Count into it, move elements 0..TopRowIndex into it, and move it over `_storage`? We get to dictate the reservation and may not run afoul of automatic grow/shrink.

Reply:
..oh, this is in ResizeTraditional, which is hardly ever used with our new defaults (ca. TH1). Hm.

Reply:
Yeah, I couldn't even get this function to execute in my playing around, so I figured if it got slower (this change definitely makes it slower), maybe it'd be ok?

Reply:
You'll _probably_ hit this if you uncheck `Wrap Text ...`:

![image](https://user-images.githubusercontent.com/189190/101299052-7da5fa00-37e5-11eb-87a7-a7b86d7aeacc.png)

Reply:
I unchecked that, made sure that I made a large buffer (more than the height) and then resized randomly a bunch.  It seems to work identically to before my change.

In both cases I see substantial memory leaks, both with and without "wrap text output on resize" checked so I will try to narrow down my repro more to file a separate issue on that.  My change doesn't affect this leak or functionality.

Reply:
Upon further investigation I guess this isn't a leak - if you resize a window wider, the CharRow objects need to grow to accomodate that, but if you shrink the window again it does not let go of that memory.  This seems like an ok way to do things to avoid thrashing as windows are resized a bunch.  Eventually once you get to the widest state you'll settle down to a steady state.

So anyway, yeah, ResizeTraditional seems fine.  Eyeball effect was that it felt roughly the same speed despite getting rid of the deque and thus doing quite a bit more work in the ResizeTraditional path to rotate the rows.  It's probably slower, but not so slow that it materially affects the user experience on my machine.

Reply:
Considering that resolved, then. Thanks!